test/unit/identity/interaction/util/EjsTemplateRenderer.test.ts

```diff
@@ -0,0 +1,19 @@
+import { renderFile } from 'ejs';
+import {
+  EjsTemplateRenderer,
+} from '../../../../../src/identity/interaction/util/EjsTemplateRenderer';
+
+jest.mock('ejs');
+
+describe('An EjsTemplateRenderer', (): void => {
+  const templatePath = '/var/templates/';
+  const templateFile = 'template.ejs';
+  const options = 'options!';
```

Comment:
I don't like this exclamation mark pattern that I've seen in a couple of tests. Can we fill out actual values in those places and then test using object equality?

That enables the unit tests to also function as an API/typical usage demo.

Reply:
Have changed some of them to make sense. The two main cases in which I kept them is in the case an `HttpResponse`/Request is expected as input since that should always be an object that got passed along anyway. Could create a Writable stream and mock that to an `HttpResponse` but in both cases it's still taking a random value and mocking it to the expected. Could use the createRequest/Response function from the mocking library if you think that would be better.

The other case is the result of calling external libraries that return complex objects of which the functionality is not really relevant, such as the `interactionDetails` call to the oidc library.

Reply:
But complex objects should just be `const interationDetails = {}` and then check for exact equality.